src/pages/question.js

```diff
@@ -0,0 +1,152 @@
+import { useEffect, useState } from "react";
+import NavBar from "../components/navBar";
+import Vector from "../assets/Vector.svg";
+import Enter from "../assets/Enter_icon.svg";
+import { useNavigate } from "react-router-dom";
+
+async function getAPIData(url) {
+  const response = await fetch(process.env.REACT_APP_API_URL + url, {
+    credentials: "same-origin",
+    headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+  });
+  const json = await response.json();
+  return json;
+}
+
+async function post(url, body) {
+  const response = await fetch(process.env.REACT_APP_API_URL + url, {
+    method: "POST",
+    body: JSON.stringify(body),
+    headers: {
+      "Content-type": "application/json; charset=UTF-8",
+      Authorization: `Bearer ${localStorage.getItem("token")}`,
+    },
+  });
+  const json = await response.json();
+  return json;
+}
+
+const getQuestion = async () => {
+  var question_id = await getAPIData("weekly_question").then(
+    (response) => response["question_id"]
+  );
+  var content = await getAPIData("questions/" + question_id.toString()).then(
+    (response) => response["content"]
+  );
+  return content;
+};
+
+function Question() {
+  const [content, setContent] = useState("");
+  const [noAnswers, setNoAnswers] = useState(true);
+
+  let navigate = useNavigate();
+
+  const [answers, setAnswers] = useState([]);
+
+  async function getAll() {
+    var question_id = await getAPIData("weekly_question").then(
+      (response) => response["question_id"]
+    );
+    const answers = await getAPIData("answers/question/" + question_id);
+    await Promise.all(
+      answers.map((answer) => {
+        return new Promise((res) => {
+          getAPIData("users/" + answer["user_id"].toString()).then((user) =>
+            res({ user, body: answer.body })
+          );
+        });
+      })
+    ).then((values) => {
+      const data = values.map((value) => {
+        setNoAnswers(false);
+        return {
+          email: value.user.email,
+          body: value.body,
+        };
+      });
+      setAnswers(data);
+    });
+  }
+
+  useEffect(() => {
+    getQuestion().then((response) => setContent(response));
```

Comment:
Right now you're calling `getAPIData("weekly_question")` 3 separate times in this component. It might be better to call it in `getQuestion` and then save it to state for the following cases

Reply:
I'm trying to do it this way and now it gives that I should add `getAll` to the useEffect and this way it's always running in a cycle. 🫤

```
const getQuestion = async () => {
  var question_id = await getAPIData("weekly_question").then(
    (response) => response["question_id"]
  );
  var content = await getAPIData("questions/" + question_id.toString()).then(
    (response) => response["content"]
  );
  return {
    question: content,
    id: question_id
  };
};

function Question() {

const [question_id, setQuestionId] = useState(0);

useEffect(() => {
    getQuestion().then((response) => {
      setContent(response.question)
      setQuestionId(response.id)
    });
    getAll();
  }, []);
```

Reply:
You could create a second useEffect that calls `getAll()` and depends on `questionId`. So in that case the first useEffect calls the `getQuestion` that sets the questionId, and that triggers the second that calls getAll. Another option would be to pass the questionId as a param to getAll